osbrain/tests/test_agent_async_requests.py

```diff
@@ -139,12 +135,12 @@ def test_unknown(nsproxy, server_client_late_reply_return):
 
     # Manually remove the pending request before it is received
     client.set_attr(_pending_requests={})
-    time.sleep(1.1)
-    assert server.get_attr('received') == ['foo']
-    assert client.get_attr('received') == []
+    assert wait_agent_attr(server, data='foo')
+    assert wait_agent_attr(client, value=[])
     assert logger_received(logger,
                            log_name='log_history_warning',
-                           message='Received response for an unknown request!')
+                           message='Received response for an unknown request!',
+                           timeout=3)
```

Comment:
Maybe it is not necessary to use `timetout=3` here. Can't we use the default?

Reply:
It was failing with the default, hence the first fixup I pushed.